airflow/dag_processing/processor.py

```diff
@@ -231,6 +231,9 @@ def _kill_process(self) -> None:
         if self._process.is_alive() and self._process.pid:
             self.log.warning("Killing DAGFileProcessorProcess (PID=%d)", self._process.pid)
             os.kill(self._process.pid, signal.SIGKILL)
+
+            # Reap the spawned zombie
+            os.waitpid(self._process.pid, 0)
```

Comment:
Indeed there might be a short time between even SIGKILL gets processed so waiting here makes sense.

Reply:
I don't know if there's anything here that could be racy, but there are some details here: https://bugs.python.org/issue42558.

Reply:
Ough.  Very interesting . So it seems that this one (or similar) is safer (ad waitpid might crash for Python 3.9+) :) 

Do I read it right ? 

```
while self._process.poll() is None:
    sleep(0.001)
```

Reply:
Yes that is how I read it too.

@pcolladosoto you wanna make that change? I don't think I had anything else that needed changing.

Reply:
Hi @malthe! I just pushed the changes: I decided to import the `time` module so that I could call `time.sleep()` within the `while` loop. Hope I made the right choice... You can check out the changes on 27502be2e2a8db81. Thanks a lot for the input! This is something that I'm sure will come in handy for future projects 😸